### ISSUE TITLE
weird and wrong visualization results

### ISSUE THREAD

**Issue by monajalal** [zju3dv/clean-pvnet#282]
Visualization of cat produces very weird results. Could you please help how to fix this?

```
(clean-pvnet) mona@ada:~/clean-pvnet$ python run.py --type evaluate --cfg_file configs/linemod.yaml model cat cls_type cat
Load model: data/model/pvnet/cat/199.pth
loading annotations into memory...
Done (t=0.20s)
creating index...
index created!
loading annotations into memory...
Done (t=0.03s)
creating index...
index created!
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1002/1002 [00:34<00:00, 28.78it/s]
2d projections metric: 0.0
ADD metric: 0.0
5 cm 5 degree metric: 0.0
mask ap70: 0.9960079840319361
(clean-pvnet) mona@ada:~/clean-pvnet$ python run.py --type linemod cls_type cat
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 177/177 [00:00<00:00, 5856.26it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10000/10000 [00:11<00:00, 908.93it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10000/10000 [00:01<00:00, 8143.13it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1002/1002 [00:00<00:00, 8749.05it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1187/1187 [00:00<00:00, 9912.89it/s]
(clean-pvnet) mona@ada:~/clean-pvnet$ python run.py --type visualize --cfg_file configs/linemod.yaml model cat cls_type cat
Load model: data/model/pvnet/cat/199.pth
loading annotations into memory...
Done (t=0.11s)
creating index...
index created!
loading annotations into memory...
Done (t=0.02s)
creating index...
index created!
  0%|▎                                                                                                                                                                   | 2/1002 [00:16<2:10:27,  7.83s/it]^C^C^C^X^Z
[2]+  Stopped                 python run.py --type visualize --cfg_file configs/linemod.yaml model cat cls_type cat

```

![Screenshot from 2023-11-20 09-45-36](https://github.com/zju3dv/clean-pvnet/assets/1892917/6e21c8e4-e0d6-48a3-8521-0d67f68edfff)
![Screenshot from 2023-11-20 09-45-43](https://github.com/zju3dv/clean-pvnet/assets/1892917/f0a53f1d-ec41-4aad-a8b5-3498d8aacd29)
![Screenshot from 2023-11-20 09-45-48](https://github.com/zju3dv/clean-pvnet/assets/1892917/cebd1115-6a51-4610-b713-cc1a5a43ed1e)

**Comment by monajalal** [comment]
fixed please check this related issue https://github.com/zju3dv/clean-pvnet/issues/283

![Screenshot from 2023-12-04 14-09-34](https://github.com/zju3dv/clean-pvnet/assets/1892917/f3444c85-4a10-480e-87d2-c2e479d90ef8)

Thanks a lot to @ZYLer0510

**Comment by utsavrai** [comment]
Hi @monajalal, I am facing the same issue and despite your suggestion to refer issue #283 I was not able to resolve the problem of incorrect bounding box.

I am using cuda 11.8, torch 2.2.0 on Ubuntu 22.04 and RTX 4060 GPU.

I have also fixed all the issue like #298 and the results  of running 
` python run.py --type evaluate --cfg_file configs/linemod.yaml model cat cls_type cat test.vsd True`
```

Load model: data/model/pvnet/cat/199.pth
loading annotations into memory...
Done (t=0.15s)
creating index...
index created!
loading annotations into memory...
Done (t=0.02s)
creating index...
index created!
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1002/1002 [00:34<00:00, 28.87it/s]
2d projections metric: 0.39520958083832336
ADD metric: 0.09181636726546906
5 cm 5 degree metric: 0.1407185628742515
mask ap70: 0.041916167664670656
```

But the when visualising 
` python run.py --type visualize --cfg_file configs/linemod.yaml model cat cls_type cat`
I get the following output

![Figure_1](https://github.com/zju3dv/clean-pvnet/assets/25121619/a20b5955-0bb1-4e70-80e1-e7a5ed79fe80)
